cucumber/src/main/java/eu/drus/jpa/unit/cucumber/BeanFactory.java

```diff
@@ -0,0 +1,41 @@
+package eu.drus.jpa.unit.cucumber;
+
+import static eu.drus.jpa.unit.cucumber.utils.ClassLoaderUtils.tryLoadClassForName;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+import eu.drus.jpa.unit.api.JpaUnitException;
+
+public final class BeanFactory {
+
+    private BeanFactory() {}
+
+    public static <T> T createBean(final Class<T> clazz) {
+        final Class<?> bpClass = tryLoadClassForName("org.apache.deltaspike.core.api.provider.BeanProvider");
+        if (bpClass != null) {
+            try {
+                final Method getContextualReference = bpClass.getMethod("getContextualReference", Class.class, Annotation[].class);
+                return clazz.cast(getContextualReference.invoke(null, clazz, new Annotation[0]));
+            } catch (final Exception e) {
+                // CDI and Deltaspike are not present and configured - fall back
+            }
+        }
+
+        // TODO: implement lookup for different DI implementations. For now only CDI is supported
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)